tables/AlternateMans/BufferedColumnarFile.h

```diff
@@ -0,0 +1,307 @@
+#ifndef CASACORE_BUFFERED_COLUMNAR_FILE_H_
+#define CASACORE_BUFFERED_COLUMNAR_FILE_H_
+
+#include <cassert>
+#include <complex>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "BitPacking.h"
+#include "RowBasedFile.h"
+
+namespace casacore {
+
+/**
+ * Class that provides binary table I/O. It is rather low-level, and requires
+ * the user to keep track of the columns and their datatype. It uses a simple
+ * binary format where the columns are interleaved into the output file.
+ * Booleans are written with bit-packing.
+ *
+ * This class limits calls to the read/write functions by caching a block of
+ * data around the last accessed row. The size of the buffer is templated to
+ * make testing of the caching easier. Production code can normally use the
+ * alias BufferedColumnarFile, which uses a reasonable buffer size.
+ *
+ * This class aims to be as fast as possible for large data files that are read
+ * from or written to consecutively. It does not try to optimize random access,
+ * partial data access and access is not transactional.
+ *
+ * The class uses exceptions to handle any I/O errors.
+ */
+template <uint64_t BufferSize = 100 * 1024>
+class VarBufferedColumnarFile : private RowBasedFile {
```

Comment:
Please add tests for this new class.

Reply:
You've already reviewed those tests ;). In the unit tests you reviewed, the tests for SimpleColumnarFile are also all executed on the BufferedColumnarFile.

Reply:
I didn't see this PR yet when reviewing the tests... There's no specific test for the BufferedColumnarFile, for example a test that explicitly fills the internal buffer only partially, but the generic tests should already provide decent coverage.